CRM/MembershipExtras/Hook/Pre/MembershipPaymentPlanProcessor/LineItem.php

```diff
@@ -17,12 +21,59 @@ public function __construct(&$params) {
    * of the line item to be inline with the new contribution amount.
    */
   public function alterLineItemParameters() {
+    if (isset($this->params['membership_type_id'])) {
+      $lineItemMembershipType = CRM_Member_BAO_MembershipType::findById($this->params['membership_type_id']);
+      //We only pro rate line item if price set is used and price field membership period type is fixed
+      if ($this->isUsingPriceSet() && $lineItemMembershipType->period_type == 'fixed') {
+        $this->calculateProRataLineItemAmounts($lineItemMembershipType, $this->membershipId);
+      }
+      else {
+        $this->calculateLineItemAmounts();
+      }
+    }
+    else {
+      $this->calculateLineItemAmounts();
+    }
+  }
+
+  /**
+   * Calcuclates line item amounts and assign amounts to line item.
+   */
+  private function calculateLineItemAmounts() {
     $this->params['line_total'] = $this->calculateSingleInstalmentAmount($this->params['line_total']);
     $this->params['unit_price'] = $this->calculateSingleInstalmentAmount($this->params['unit_price']);
-
     if (!empty($this->params['tax_amount'])) {
       $this->params['tax_amount'] = $this->calculateSingleInstalmentAmount($this->params['tax_amount']);
     }
   }
 
+  /**
+   * Calculates pro rata amounts for line item or fixed period membership type
+   * base on payment plan schedule, membreship type, and membership start date.
+   * *
```

Comment:
Should change this comment to `Calculates pro rata amounts for line item for fixed period membership type`.